Dockerfile.base

```diff
@@ -0,0 +1,27 @@
+# This is the base image used for Coder images. It's a multi-arch image that is
```

Comment:
Why do we need a base? Can we do all builds in depot? then we don't need the forking logic.

Reply:
Because then every developer needs to have depot access and install depot CLI and authenticate it just to build docker images. The forking logic makes it easier overall as most people won't have to worry about dealing with the build process anyways.

I also don't think it adds that much overhead and it's clearly documented at the top of each Dockerfile what they're for.

Reply:
Makes sense!